### PR TITLE
feat(@angular/cli): allow component css imports

### LINKED ISSUE #4285: CSS imports do not work for component CSS
In `beta.26`, the following results in a runtime error:
```
// src/app/app.component.css
@import '~bootstrap/scss/bootstrap'
```
```
http://localhost:4200/~bootstrap/scss/bootstrap Failed to load resource: the server responded with a status of 404 (Not Found)
polyfills.bundle.js:6292 Unhandled Promise rejection: Failed to load ~bootstrap/scss/bootstrap ; Zone: <root> ; Task: Promise.then ; Value: Failed to load ~bootstrap/scss/bootstrap 
```

For global styles, `css-loader` is what resolves imports currently. Component styles had `css-loader` added in an incorrect way, which resulted in broken CSS (fixed by @clydin in https://github.com/angular/angular-cli/pull/3511).

Using `'exports-loader?module.exports.toString()'` together with `css-loader` fixes this bug, but I know @clydin is currently working in `css-loader` replacement functionality via PostCSS plugins for imports, which also solves the problem so it's better to wait for that solution.

### PR DESCRIPTION
**Pull request by filipesilva**
Fix #4285

### CODE REVIEW

File: packages/@angular/cli/models/webpack-configs/styles.ts [hunk 1]
```diff
@@ -86,21 +86,27 @@ export function getStylesConfig(wco: WebpackConfigOptions) {
...
+    // so we need to add options in its query
+    `css-loader?${JSON.stringify({ sourceMap: cssSourceMap })}`,
+    'postcss-loader'
+  ];
 
   // load component css as raw strings
   let rules: any = baseRules.map(({test, loaders}) => ({
-    exclude: globalStylePaths, test, loaders: ['raw-loader', ...commonLoaders, ...loaders]
+    exclude: globalStylePaths, test, loaders: [
+      'exports-loader?module.exports.toString()',
```
Comment (clydin) [inline packages/@angular/cli/models/webpack-configs/styles.ts]: You'll need a comment with `require('exports-loader')` at the top of the file with the others.

Comment (filipesilva) [inline packages/@angular/cli/models/webpack-configs/styles.ts]: Done

File: packages/@angular/cli/models/webpack-configs/styles.ts [hunk 2]
```diff
@@ -86,21 +86,27 @@ export function getStylesConfig(wco: WebpackConfigOptions) {
     }
   ];
 
-  const commonLoaders = ['postcss-loader'];
+  const commonLoaders = [
+    // css-loader doesn't support webpack.LoaderOptionsPlugin properly,
+    // so we need to add options in its query
+    `css-loader?${JSON.stringify({ sourceMap: cssSourceMap })}`,
```
Comment (filipesilva) [inline packages/@angular/cli/models/webpack-configs/styles.ts]: Done

**Comment by adam-beck** [comment]
I ran the eject function and noticed this loader being used. I'm perplexed on how it is working and would really appreciate some enlightenment. I went through the [`exports-loader`](https://github.com/webpack-contrib/exports-loader/blob/master/index.js) and it looks like the following line would be added to the CSS files:

```
module.exports = module.exports.toString()
```

I understand the CSS file is now a module but what does `module.exports.toString()` output. I'm not sure how to find that out and why it would work. Also, what is the point of using CSS modules when Angular has its own "modules" via style encapsulation? Are there any benefits or tradeoffs to be made aware of? How would this even be used? Something like:

```
import styles from './component.styles.css';

...

template: `<div class=${styles.greeting}>Hello, World!</div>`
```

Thank you for your time.

**Pull request merged**